src/Storages/MergeTree/ReplicatedMergeMutateTaskBase.cpp

```diff
@@ -69,11 +69,11 @@ bool ReplicatedMergeMutateTaskBase::executeStep()
             else
                 tryLogCurrentException(log, __PRETTY_FUNCTION__);
 
-            /** This exception will be written to the queue element, and it can be looked up using `system.replication_queue` table.
-                 * The thread that performs this action will sleep a few seconds after the exception.
-                 * See `queue.processEntry` function.
-                 */
-            throw;
+            /// This exception will be written to the queue element, and it can be looked up using `system.replication_queue` table.
+            /// The thread that performs this action will sleep a few seconds after the exception.
+            /// See `queue.processEntry` function.
+            if (!retryable_error)
+                throw;
```

Comment:
Unfortunately, it will not work because we have some logic built on top of exceptions...

Reply:
Btw, I was thinking that maybe we have some backoff on retries of failed queue entries, it would help too

Reply:
>Unfortunately, it will not work because we have some logic built on top of exceptions... (so we have to rethow it unconditionally, otherwise replication queue may get stuck, as you can see in the failed tests)

Yeah, now I see, though it is not about replication queue, but about abort in the WriteBuffer dtor, but maybe there are more problems?

Anyway, I rewrote the patch, introduced `IExecutableTask::printExecutionException()` to let the `MergeTreeBackgroundExecutor` decide. Looks a little bit hackish I too much, maybe, but what do you think?